machine/v1beta1/types_awsprovider.go

```diff
@@ -17,6 +17,9 @@ type AWSMachineProviderConfig struct {
 	AMI AWSResourceReference `json:"ami"`
 	// instanceType is the type of instance to create. Example: m4.xlarge
 	InstanceType string `json:"instanceType"`
+	// cpuOptions is the set of cpu options for the instance, where you can enable/disable AMD SEV-SNP on the instance.
+	// +optional
+	CpuOptions *CpuOptions `json:"cpuOptions,omitempty"`
 	// tags is the set of tags to add to apply to an instance, in addition to the ones
```

Comment:
Hi @fangge1212, would it make sense to use a field named `ConfidentialComputing` (or something similar) that could be `Disabled` or `AmdSevSnp` instead?

In case AWS supports other confidential computing technologies in the future, that would be easier to extend than this approach IMO. As it would require adding another new field to the API instead of a new value to the enum.

Reply:
Same as https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/5598, waiting for more opinions before chaning it